snapshots/manager_test.go

```diff
@@ -79,7 +79,7 @@ func TestManager_Take(t *testing.T) {
 		Height: 5,
 		Format: snapshotter.SnapshotFormat(),
 		Chunks: 1,
-		Hash:   []uint8{0xcd, 0x17, 0x9e, 0x7f, 0x28, 0xb6, 0x82, 0x90, 0xc7, 0x25, 0xf3, 0x42, 0xac, 0x65, 0x73, 0x50, 0xaa, 0xa0, 0x10, 0x5c, 0x40, 0x8c, 0xd5, 0x1, 0xed, 0x82, 0xb5, 0xca, 0x8b, 0xe0, 0x83, 0xa2},
```

Comment:
Why does this `Hash` value depend on the go version? Is it mean that snapshots created on go 1.18 nodes and transferred to go 1.19 nodes cannot be restored correctly? Is it the same as the problem for AppHash?

Reply:
I have even confirmed that this hash is from:
https://github.com/Finschia/finschia-sdk/blob/main/snapshots/store.go#L259

Reply:
The difference in this `Hash` value between go versions was due to differences in the binaries generated by the compression of the snapshot. By removing the compressed output stream layer from `StreamWriter`, which is responsible to generate the binaries, as follows, it was confirmed that the same binaries are generated and the hash values matched for both 1.18 and 1.19.

<img width="927" alt="StreamWriter with compression process removed" src="https://github.com/Finschia/finschia-sdk/assets/836654/7c6bbcd3-ecb7-4926-8532-ee6433fb9ad3">

It makes sense for snapshots to be compressed and have a hash value for checksum purposes to detect any loss during transmission. But in this test case, it should be verified that the binary SHA-256 value for the generated snapshot matches the `Hash` value, rather than hard-coding a `Hash` value that depends on the implementation and behaviour of the compression algorithm. This could be done in a separate PR and should be done for the Cosmos SDK.

In this test case, the `Hash` value is found to match the value generated by SHA-256 for concatenated `expectChunks`.

<img width="825" alt="2023-06-07 14 46 58" src="https://github.com/Finschia/finschia-sdk/assets/836654/3c1e875e-9b69-459f-aa2f-da9ea692a2c0">

Reply:
I also think that it is appropriate to conduct this verification by PR for the main branch of finschia-sdk or the original cosmos-sdk.(Even if you submit a PR to the head office, it may take some time for finschia-sdk to backport the version, so it may be safer to submit a PR to our main branch.)

In any case, in this PR, I think that the hash value change due to version update of go can be left as it is.

Reply:
The relevant fixes aren't the purpose of this PR and should be a separate PR.

The root cause of the different compression binary is explained in the following cosmos-sdk issue: before 1.18, the `Close()` is called twice to add two trailers, and after 1.19 the second `Close()` is ignored and only one trailer is added.
https://github.com/cosmos/cosmos-sdk/issues/13373